huaweicloud/common/common.go

```diff
@@ -50,6 +51,17 @@ func CheckDeleted(d *schema.ResourceData, err error, msg string) error {
 	return fmtp.Errorf("%s: %s", msg, err)
 }
 
+// CheckDeletedDiag checks the error to see if it's a 404 (Not Found) and, if so,
+// sets the resource ID to the empty string instead of throwing an error.
+func CheckDeletedDiag(d *schema.ResourceData, err error, msg string) diag.Diagnostics {
+	if _, ok := err.(golangsdk.ErrDefault404); ok {
+		d.SetId("")
+		return nil
+	}
+
+	return diag.Errorf("%s: %s", msg, err)
+}
+
```

Comment:
The name of the function is best to clearly indicate the main effect

Reply:
just keep the same behavior as `common.CheckDeleted`